src/net_processing.cpp

```diff
@@ -3177,6 +3177,13 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             return;
         }
 
+        if (pfrom.IsFullRBF() && !HasFullRBFServiceFlag(nServices))
+        {
+            LogPrint(BCLog::NET, "peer=%d does not offer fullrbf as expected; disconnecting\n", pfrom.GetId());
```

Comment:
According to the current code, this is impossible? Because the only way to become `IsFullRBF()` is if the latter condition is true.
In that case, should it be ` Assume` instead?